src/itwinai/torch/tuning.py

```diff
@@ -0,0 +1,158 @@
+from ray.tune.schedulers import (
+    AsyncHyperBandScheduler,
+    HyperBandForBOHB,
+    HyperBandScheduler,
+    PopulationBasedTraining,
+)
+from ray.tune.schedulers.pb2 import PB2  # Population Based Bandits
+from ray.tune.search.bayesopt import BayesOptSearch
+from ray.tune.search.bohb import TuneBOHB
+from ray.tune.search.hyperopt import HyperOptSearch
+
+
+def get_raytune_search_alg(
+    tune_config, seeds=False
+) -> TuneBOHB | BayesOptSearch | HyperOptSearch | None:
+    """Get the appropriate Ray Tune search algorithm based on the provided configuration.
+
+    Args:
+        tune_config (Dict): Configuration dictionary specifying the search algorithm,
+            metric, mode, and, depending on the search algorithm, other parameters.
+        seeds (bool, optional): Whether to use a fixed seed for reproducibility for some
+            search algorithms that take a seed. Defaults to False.
+
+    Returns:
+        An instance of the chosen Ray Tune search algorithm or None if no search algorithm is
+            used or if the search algorithm does not match any of the supported options.
+
+    Notes:
+        - `TuneBOHB` is automatically chosen for BOHB scheduling.
+    """
+    if "scheduler" in tune_config:
+        scheduler = tune_config["scheduler"]["name"]
+    else:
+        scheduler = None
```

Comment:
Ironically, here you can use `.get()` to avoid some lines

Reply:
Okay I've changed this, I find it a bit more convoluted though tbh, what do you think?
`scheduler = tune_config.get("scheduler", {}).get("name")`

Reply:
I agree with you, it was better the way it was before. I missed the fact that you have the double-dictionary structure. That does beg the question though: Why is it called "scheduler" when you have to extract the "name" from the scheduler? Maybe the variable should be called "scheduler_name" or something instead? Not super important haha

Reply:
Hmm I see your point, but there is a few more variables that need to be defined under "scheduler", it could of course be named something else, but the value of "scheduler" needs to be a dictionary. I will resolve this because we talked about it in-person :)